netlogo-gui/src/main/app/WidgetActions.scala

```diff
@@ -0,0 +1,145 @@
+package org.nlogo.app
+
+import java.awt.Rectangle
+import java.awt.event.ActionEvent
+import javax.swing.{AbstractAction, JComponent}
+import javax.swing.undo.{AbstractUndoableEdit, UndoManager, UndoableEdit}
+
+import org.nlogo.core.Widget
+import org.nlogo.swing.ToolBarButton
+import org.nlogo.window.MouseMode
+
+object WidgetActions {
+
+  val undoManager = new org.nlogo.editor.UndoManager() {
+    // The default one doesn't work!
+    override def canRedo: Boolean = {
+      if(editToBeRedone() != null) {
+        return true
+      }
+      false
+    }
+  }
+
+  def addWidget(widgetPanel: WidgetPanel, coreWidget: Widget, x: Int, y: Int): Unit ={
```

Comment:
Space between `=` and `{` (throughout this file, also in `WidgetPanel.reAddWidget`).